redaxo/src/addons/phpmailer/lib/mailer.php

```diff
@@ -505,4 +505,136 @@ protected function microsoft365Send(): bool
+        $finder = rex_finder::factory($archiveFolder)->recursive()->filesOnly();
+
+        foreach ($finder as $file) {
+            $archiveSize += $file->getSize();
+            if ('eml' === pathinfo($file->getFilename(), PATHINFO_EXTENSION)) {
+                ++$fileCount;
+            }
+        }
+
+        return ['size' => $archiveSize, 'fileCount' => $fileCount];
+    }
+
+    /**
+     * Get recent archived email files.
+     *
+     * @param int $limit Maximum number of files to return
+     * @return array<string> Array of file paths
+     */
+    public static function getRecentArchivedFiles(int $limit = 10): array
+    {
+        $archiveFolder = self::logFolder();
+
+        if (!is_dir($archiveFolder)) {
+            return [];
+        }
+
+        // Get recent .eml files recursively using rex_finder
+        $finder = rex_finder::factory($archiveFolder)->recursive()->filesOnly();
+        $files = [];
+
+        foreach ($finder as $path => $file) {
+            if ('eml' === pathinfo($file->getFilename(), PATHINFO_EXTENSION)) {
+                $files[] = $path;
+            }
+        }
+
+        if (empty($files)) {
+            return [];
+        }
+
+        // Sort by modification time, newest first
+        usort($files, static function (string $a, string $b): int {
+            $timeA = filemtime($a);
+            $timeB = filemtime($b);
+            if (false === $timeA || false === $timeB) {
+                return 0;
+            }
+            return $timeB - $timeA;
+        });
+
+        // Return only the requested number of files
+        return array_slice($files, 0, $limit);
+    }
+
+    /**
+     * Parse email headers from .eml file.
+     *
+     * @param string $filePath Path to .eml file
+     * @return array{subject: string, recipient: string, size: int, mtime: int}
+     */
+    public static function parseEmailHeaders(string $filePath): array
+    {
+        $addon = rex_addon::get('phpmailer');
+        $subject = $addon->i18n('archive_no_subject');
+        $recipient = $addon->i18n('archive_no_recipient');
+        $filesize = filesize($filePath);
+        $filemtime = filemtime($filePath);
+
+        // Use rex_file::get() to read file content
+        $content = rex_file::get($filePath);
+        if ($content) {
+            // Split content into lines and process only headers
+            $lines = explode("\n", $content);
+            $headerLines = 0;
+
+            foreach ($lines as $line) {
+                $line = trim($line);
+                if (empty($line)) {
+                    break; // End of headers
+                }
+
+                if ($headerLines >= 50) {
+                    break; // Limit header processing
+                }
+
+                if (0 === stripos($line, 'Subject:')) {
+                    $subject = substr($line, 8);
+                    // Decode MIME encoded subjects
+                    if (function_exists('iconv_mime_decode')) {
```

Comment:
iconv ist Pflichtextension (wird in setup geprüft). daher kein function_exists notwendig.